### ISSUE TITLE
Gemfile infinite loop during rake bootstrap

### ISSUE THREAD

**Issue by rchinsio-gpsw** [CocoaPods/Rainforest#53]
I'm trying to get the development environment setup. I had the same issue as this https://github.com/CocoaPods/Rainforest/issues/38 so I installed Ruby 2.1.2 and now I get an error about gems depending on each other and creating an infinite loop.

```
--------------------------------------------------------------------------------
Environment bootstrap
--------------------------------------------------------------------------------

Updating submodules
/Users/rchinsio/.rbenv/versions/2.1.2/bin/bundle
Installing gems

Bootstrapping cocoapods-deintegrate
rake aborted!
Bundler::GitError: Cannot use local override for CocoaPods at /Users/rchinsio/git/Rainforest/CocoaPods Bundler::GitError: Cannot use local override for CocoaPods at /Users/rchinsio/git/Rainforest/CocoaPods because :branch is not specified in Gemfile. Specify a branch or use `bundle config --delete` to remove the local override
/Users/rchinsio/git/Rainforest/cocoapods-deintegrate/Rakefile:2:in `<top (required)>'
(See full trace by running task with --trace)

Bootstrapping cocoapods-docs
rake --no-search bootstrap
/Users/rchinsio/.rbenv/versions/2.1.2/bin/bundle

--------------------------------------------------------------------------------
Installing gems
--------------------------------------------------------------------------------

bundle install
Fetching gem metadata from http://rubygems.org/............
Fetching version metadata from http://rubygems.org/...
Fetching dependency metadata from http://rubygems.org/..
Resolving dependencies...
Your Gemfile requires gems that depend on each other, creating an infinite loop. Please remove either gem 'cocoapods' or gem 'cocoapods-deintegrate' and try again.
rake aborted!
Command failed with status (21): [bundle install...]
/Users/rchinsio/git/Rainforest/cocoapods-docs/Rakefile:9:in `block in <top (required)>'
Tasks: TOP => bootstrap
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [rake --no-search bootstrap...]
/Users/rchinsio/git/Rainforest/Rakefile:94:in `block (3 levels) in <top (required)>'
/Users/rchinsio/git/Rainforest/Rakefile:92:in `chdir'
/Users/rchinsio/git/Rainforest/Rakefile:92:in `block (2 levels) in <top (required)>'
/Users/rchinsio/git/Rainforest/Rakefile:91:in `each'
/Users/rchinsio/git/Rainforest/Rakefile:91:in `block in <top (required)>'
/Users/rchinsio/git/Rainforest/Rakefile:53:in `block in <top (required)>'
Tasks: TOP => bootstrap_repos
(See full trace by running task with --trace)
```

**Comment by segiddins** [comment]
The local override there is probably the issue

**Comment by rchinsio-gpsw** [comment]
I didn't do anything other than git clone and rake bootstrap so that file is what came straight from git.

Am I supposed to be modifying the Gemfile?

**Comment by segiddins** [comment]
```
Bundler::GitError: Cannot use local override for CocoaPods at /Users/rchinsio/git/Rainforest/CocoaPods Bundler::GitError: Cannot use local override for CocoaPods at /Users/rchinsio/git/Rainforest/CocoaPods because :branch is not specified in Gemfile. Specify a branch or use `bundle config --delete` to remove the local override
/Users/rchinsio/git/Rainforest/cocoapods-deintegrate/Rakefile:2:in `<top (required)>
```
Bundler is saying you have a local override setup

**Comment by rchinsio-gpsw** [comment]
I removed ~/.bundle/config which had some generated cocoapods stuff from a previous Rainforest command?

Anyway, now if I run rake bootstrap I get a different error:

```
--------------------------------------------------------------------------------
Environment bootstrap
--------------------------------------------------------------------------------

Updating submodules
/Users/rchinsio/.rbenv/versions/2.1.2/bin/bundle
Installing gems

Bootstrapping cocoapods-deintegrate
rake aborted!
Bundler::GemNotFound: Could not find minitest-5.8.2 in any of the sources
/Users/rchinsio/git/Rainforest3/cocoapods-deintegrate/Rakefile:2:in `<top (required)>'
(See full trace by running task with --trace)

Bootstrapping cocoapods-docs
rake --no-search bootstrap
/Users/rchinsio/.rbenv/versions/2.1.2/bin/bundle

--------------------------------------------------------------------------------
Installing gems
--------------------------------------------------------------------------------

bundle install
Updating https://github.com/CocoaPods/CocoaPods.git
Updating https://github.com/CocoaPods/Core.git
Updating https://github.com/CocoaPods/CLAide.git
Fetching gem metadata from http://rubygems.org/............
Fetching version metadata from http://rubygems.org/...
Fetching dependency metadata from http://rubygems.org/..
Resolving dependencies...
Your Gemfile requires gems that depend on each other, creating an infinite loop. Please remove either gem 'cocoapods' or gem 'cocoapods-deintegrate' and try again.
rake aborted!
Command failed with status (21): [bundle install...]
/Users/rchinsio/git/Rainforest3/cocoapods-docs/Rakefile:9:in `block in <top (required)>'
Tasks: TOP => bootstrap
(See full trace by running task with --trace)
rake aborted!
Command failed with status (1): [rake --no-search bootstrap...]
/Users/rchinsio/git/Rainforest3/Rakefile:94:in `block (3 levels) in <top (required)>'
/Users/rchinsio/git/Rainforest3/Rakefile:92:in `chdir'
/Users/rchinsio/git/Rainforest3/Rakefile:92:in `block (2 levels) in <top (required)>'
/Users/rchinsio/git/Rainforest3/Rakefile:91:in `each'
/Users/rchinsio/git/Rainforest3/Rakefile:91:in `block in <top (required)>'
/Users/rchinsio/git/Rainforest3/Rakefile:53:in `block in <top (required)>'
Tasks: TOP => bootstrap_repos
(See full trace by running task with --trace)
```

**Comment by igagis** [comment]
same thing here

**Comment by protikhonov** [comment]
I have the same thing. Does anyone know how to fix this issue?

**Comment by amake** [comment]
I have similar problems, culminating in the same "infinite loop" message, even installing in a fresh VM. Log here:
http://pastebin.com/A0j9Cp3y

**Comment by segiddins** [comment]
Thanks, we'll try and look into this!

**Comment by mr-fixit** [comment]
Still happening...

**Comment by arifken** [comment]
I ran into this issue trying to include CocoaPods built from the master branch from a Gemfile. What I found as the culprit is: https://github.com/CocoaPods/cocoapods-deintegrate/blob/v0.2.1/cocoapods_deintegrate.gemspec#L19

This was fixed in https://github.com/CocoaPods/cocoapods-deintegrate/commit/459ba6d4e7e04916c4ea1e9e622312916a0761ca but has not yet been released. I suspect that once it is integrated into a release of cocoapods-deintegrate, and that this new version is set in cocoapods.gemspec, that this error will no longer occur. thoughts?
